peer/network.go

```diff
@@ -365,15 +365,16 @@ func (n *network) AppRequest(ctx context.Context, nodeID ids.NodeID, requestID u
 // If the response handler returns an error it is propagated as a fatal error.
 func (n *network) AppResponse(ctx context.Context, nodeID ids.NodeID, requestID uint32, response []byte) error {
 	n.lock.Lock()
```

Comment:
since `closed` is atomic, could we move this lock to directly before
`	handler, exists := n.markRequestFulfilled(requestID)`?

Should we move it to inside `markRequestFulfilled` instead?

Reply:
Yeah that makes much more sense than what I was currently doing.